src/test/java/com/waverleysoftware/data/DataSuppliers.java

```diff
@@ -32,6 +33,20 @@ public StreamEx getSensorData() {
                 new Sensor(3, "Garage", EMBEDDED)));
     }
 
+    @SuppressWarnings("unchecked")
+    @DataSupplier(transpose = true)
+    public <T> T[] getDataCollection(final Method method) {
+        final Optional<Data[]> annotationsByType = ofNullable(method.getAnnotationsByType(Data.class));
+        if (!annotationsByType.isPresent()) {
```

Comment:
it makes no sense to perform additional check and throwing an error here... if there're no annotations found, just return an empty array... you can chain optional with below stream operations...